src/runtime_src/core/tools/xbmgmt2/flash/firmware_image.cpp

```diff
@@ -532,6 +671,28 @@ firmwareImage::firmwareImage(const char *file, imageType type) :
             in_file.seekg(bmcSection->m_sectionOffset + bmc->m_offset);
             in_file.read(mBuf, bufsize);
         }
+        else if (type == STRIPPED_FIRMWARE)
+        {
+            std::shared_ptr<char> full(new char[ap->m_header.m_length]);
```

Comment:
This will result in a mismatched free, plus it is not shared ownership.  Use `std::vector<char> full( ... length ...`);